bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,463 @@
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding shade down: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding shade down", e);
+        }
+    }
+
+    public void commandShadeStop(String targetId) throws SomfyMyLinkException {
+        try {
+            String command = buildShadeCommand("mylink.move.stop", targetId);
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding shade stop: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding shade stop", e);
+        }
+    }
+
+    public void commandScene(Integer sceneId) throws SomfyMyLinkException {
+        try {
+            String command = buildSceneCommand("mylink.scene.run", sceneId);
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding scene run: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding scene run", e);
+        }
+    }
+
+    private void sendCommand(String command) throws SomfyMyLinkException {
+        synchronized (CONNECTION_LOCK) {
+            try {
+                logger.debug("Sending: {}", command);
+                try (Socket socket = getConnection(); OutputStream out = socket.getOutputStream()) {
+                    byte[] sendBuffer = command.getBytes(StandardCharsets.US_ASCII);
+                    // send the command
+                    out.write(sendBuffer, 0, sendBuffer.length);
+                }
+
+                // give time for mylink to process
+                Thread.sleep(CONNECTION_DELAY);
+            } catch (SocketTimeoutException e) {
+                logger.warn("Timeout sending command to mylink: {} Message: {}", command, e.getMessage());
+                throw new SomfyMyLinkException("Timeout sending command to mylink", e);
+            } catch (IOException e) {
+                logger.warn("Problem sending command to mylink: {} Message: {}", command, e.getMessage());
+                throw new SomfyMyLinkException("Problem sending command to mylink", e);
+            } catch (InterruptedException e) {
+                logger.debug("Interrupted while waiting after sending command to mylink: {} Message: {}", command,
+                        e.getMessage());
+            }
+        }
+    }
+
+    private @Nullable SomfyMyLinkResponseBase sendCommandWithResponse(String command, Type responseType)
+            throws SomfyMyLinkException {
+        synchronized (CONNECTION_LOCK) {
+            try {
+                logger.debug("Sending: {}", command);
+                try (Socket socket = getConnection();
+                        OutputStream out = socket.getOutputStream();
+                        BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
```

Comment:
The default charset varies from system to system so it is better to specify it yourself instead of using the default.

Reply:
done.